### PR TITLE
Stop! AK + Jakt Time! (part 1 of N)

### PR DESCRIPTION
**Pull request by alimpfard**
Port over all the (useful) extensions made in Jakt.
This is a requirement for https://github.com/SerenityOS/jakt/pull/1284.

### CODE REVIEW

File: AK/Error.h [hunk 1]
```diff
@@ -81,9 +81,33 @@ class [[nodiscard]] ErrorOr {
...
+    ALWAYS_INLINE ErrorOr(ErrorOr<U, ErrorType>&& value)
+        : m_value_or_error(value.visit([](U& v) { return Variant<T, ErrorType>(move(v)); }, [](ErrorType& error) { return Variant<T, ErrorType>(move(error)); }))
+    {
+    }
+
     template<typename U>
     ALWAYS_INLINE ErrorOr(U&& value)
-    requires(!IsSame<RemoveCVReference<U>, ErrorOr<T, ErrorType>>)
+    requires(
+        requires { T(declval<U>()); } || requires { ErrorType(declval<RemoveCVReference<U>>()); })
```
Comment (ADKaster) [inline AK/Error.h:110]: How does this construct prevent selecting this constructor over the defaulted move ctor?

**Review by ADKaster — COMMENTED** [review]
One comment, otherwise LGTM

File: AK/Error.h (hunk 1 above)
Comment (alimpfard) [reply AK/Error.h:120]: The move ctor is always selected before this as that one isn't templated.
However, that said, you can't make a T or an ErrorType from the ErrorOr.

**Comment by alimpfard** [comment]
> Commit is a git merge commit, use the rebase command instead

wat.

File: AK/Error.h [hunk 2]
```diff
@@ -37,6 +37,13 @@ class Error {
         return from_string_view(StringView { string_literal, N - 1 });
     }
 
+    // Note: Don't call this from C++; it's here for Jakt interop.
+    template<SameAs<StringView> T>
+    ALWAYS_INLINE static Error from_string_literal(T string)
```
Comment (konradekk) [inline AK/Error.h:42]: maybe it should be name it accordingly to discourage the use? 🤔
e.g. `__jakt__from_string_literal` `__INTERNAL__from_string_literal` or similarly? (same in other places)

Comment (alimpfard) [reply AK/Error.h:42]: The name must be _exactly_ that I'm afraid (Jakt generates Foo::from_string_literal)

Comment (alimpfard) [reply AK/Error.h:42]: Oh shit sorry, this is the Error one
Yeah I guess we can name it weirdly

**Review by awesomekling — APPROVED** [review]

**Pull request merged**